src/sentry/workflow_engine/migration_helpers/issue_alert_migration.py

```diff
@@ -0,0 +1,222 @@
+import logging
+from dataclasses import dataclass
+from typing import Any
+
+from sentry.grouping.grouptype import ErrorGroupType
+from sentry.models.rule import Rule
+from sentry.rules.conditions.event_frequency import EventUniqueUserFrequencyConditionWithConditions
+from sentry.rules.processing.processor import split_conditions_and_filters
+from sentry.workflow_engine.migration_helpers.issue_alert_conditions import (
+    create_event_unique_user_frequency_condition_with_conditions,
+    translate_to_data_condition,
+)
+from sentry.workflow_engine.migration_helpers.rule_action import (
+    build_notification_actions_from_rule_data_actions,
+)
+from sentry.workflow_engine.models import (
+    AlertRuleDetector,
+    AlertRuleWorkflow,
+    DataCondition,
+    DataConditionGroup,
+    DataConditionGroupAction,
+    Detector,
+    DetectorWorkflow,
+    Workflow,
+    WorkflowDataConditionGroup,
+)
+from sentry.workflow_engine.models.data_condition import (
+    Condition,
+    enforce_data_condition_json_schema,
+)
+
+logger = logging.getLogger(__name__)
+
+SKIPPED_CONDITIONS = [Condition.EVERY_EVENT]
+
+
+@dataclass
+class IssueAlertMigrator:
+    def __init__(self, rule: Rule, user_id: int | None = None, is_dry_run: bool | None = False):
+        self.rule = rule
+        self.user_id = user_id
+        self.is_dry_run = is_dry_run
+        self.data = rule.data
+        self.project = rule.project
+        self.organization = self.project.organization
+
+    def run(self) -> None:
+        error_detector = self._create_detector_lookup()
+        conditions, filters = split_conditions_and_filters(self.data["conditions"])
+        workflow = self._create_workflow_and_lookup(
+            conditions=conditions,
+            filters=filters,
+            action_match=self.data["action_match"],
+            detector=error_detector,
+        )
+        if_dcg = self._create_if_dcg(
+            filter_match=self.data.get("filter_match", "all"),
+            workflow=workflow,
+            conditions=conditions,
+            filters=filters,
+        )
+        self._create_workflow_actions(if_dcg=if_dcg, actions=self.data["actions"])
+
+    def _create_detector_lookup(self) -> Detector:
+        if self.is_dry_run:
+            created = True
+            error_detector = Detector.objects.filter(
+                type=ErrorGroupType.slug, project=self.project
+            ).first()
+            if error_detector:
+                created = not AlertRuleDetector.objects.filter(
+                    detector=error_detector, rule=self.rule
+                ).exists()
+            else:
+                error_detector = Detector(type=ErrorGroupType.slug, project=self.project)
```

Comment:
this logic is a bit messy, but basically we want to know whether we would create a new `AlertRuleDetector` lookup mapping or not. if we don't, then we've already migrated this `Rule` and don't need to do it again

Reply:
i also have to create a temporary `Detector` object to return